pkg/core/block/block.go

```diff
@@ -221,6 +221,11 @@ func (b *Block) GetExpectedBlockSizeWithoutTransactions(txCount int) int {
 	return size
 }
 
+// GetExpectedHeaderSize returns the expected header size with empty witness.
+func (b *Block) GetExpectedHeaderSize() int {
+	return expectedHeaderSizeWithEmptyWitness
+}
```

Comment:
Nope, this is not the header size. You need GetExpectedBlockSizeWithoutTransactions except the number of transactions. And this method should be defined on Header likely.